packit/config/job_config.py

```diff
@@ -17,16 +17,30 @@ class JobType(Enum):
     """Type of the job used by users in the config"""
 
     propose_downstream = "propose_downstream"
-    build = "build"
+    pull_from_upstream = "pull_from_upstream"
+    build = "build"  # deprecated
     sync_from_downstream = "sync_from_downstream"
     copr_build = "copr_build"
-    production_build = "production_build"  # upstream koji build
+    production_build = "production_build"  # deprecated
+    upstream_koji_build = "upstream_koji_build"
     koji_build = "koji_build"  # downstream koji build
     tests = "tests"
     bodhi_update = "bodhi_update"
     vm_image_build = "vm_image_build"
 
 
+DEPRECATED_JOB_TYPES = {
+    JobType.build: "The `build` job type aimed to be an alias when Packit supported just one "
+    "build type. "
+    "There are currently more types of builds and just `build` can be misleading. "
+    "Please, be explicit and use `copr_build` instead.",
+    JobType.production_build: "The `production_build` name for upstream Koji build is misleading "
+    "when we have also another job (`koji_build`) to run downstream Koji "
+    "builds that are more production than the upstream ones. "
```

Comment:
It took me a while to realize what you mean by 'more production'.
```suggestion
    "because it's actually a scratch (i.e. non-production) build. There's "
    "the `koji_build` job for a non-scratch/production build.
```

Reply:
It's tricky because we actually support non-scratch builds from upstream as well, but no one uses that because it does not work out of the box. And for downstream, we can run both scratch and non-scratch ones... So it's not much about `production/non-scratch`.

I'll try to reword this. Thanks for the suggestion.

Reply:
@jpopelka Can you please take another look? I am still struggling with the wording but I hope it's at least a bit clearer...